Samples/AudioCreation/cs/AudioCreation/Scenario7_RecordPlayback.xaml.cs

```diff
@@ -0,0 +1,570 @@
+
+            if (requestStartPlaying)
+            {
+                Debug.Assert(!isRecording);
+                Debug.Assert(!requestStartRecording);
+                Debug.Assert(!isPlaying);
+
+                isPlaying = true;
+                requestStartPlaying = false;
+                currentIndex = 0;
+            }
+
+            if (isPlaying)
+            {
+                Debug.Assert(!isRecording);
+                Debug.Assert(!requestStartPlaying);
+                Debug.Assert(!requestStartRecording);
+
+                HandleOutgoingAudio(requiredSamples);
+
+                /*
+                int currentStatusMsec = DateTime.Now.Millisecond;
+                // update status every tenth of a second
+                if (lastStatusMsec % 100 > currentStatusMsec % 100)
+                {
+                    CoreApplication.MainView.CoreWindow.Dispatcher.TryRunAsync(
+                        Windows.UI.Core.CoreDispatcherPriority.Normal,
+                        UpdateStatusWhilePlaying);
+                }
+                lastStatusMsec = currentStatusMsec;
+                */
+            }
+        }
+
+        // We keep a 1MB AudioFrame and reuse it.
+        static AudioFrame s_audioFrame = new Windows.Media.AudioFrame(1024 * 1024);
+        private int s_zeroByteOutgoingFrameCount;
+
+        /// <summary>
+        /// Handle a frame of outgoing audio to the output device.
+        /// </summary>
+        /// <remarks>
+        /// This pushes a megabyte of audio each time, to greatly reduce the number of callbacks.
+        /// 
+        /// We don't need the requiredSamples argument but we don't 
+        /// </remarks>
+        private unsafe void HandleOutgoingAudio(uint requiredSamples)
+        {
+            if (requiredSamples == 0)
+            {
+                s_zeroByteOutgoingFrameCount++;
+                return;
+            }
+
+            using (AudioBuffer buffer = s_audioFrame.LockBuffer(AudioBufferAccessMode.Write))
+            using (IMemoryBufferReference reference = buffer.CreateReference())
+            {
+                byte* dataInBytes;
+                uint capacityInBytes;
+
+                // Get the buffer from the AudioFrame
+                ((IMemoryBufferByteAccess)reference).GetBuffer(out dataInBytes, out capacityInBytes);
+
+                uint bytesRemaining = capacityInBytes;
+                uint position = 0;
+
+                while (bytesRemaining > 0)
+                {
+                    uint bytesInThisIteration = bytesRemaining;
+                    if (currentIndex + bytesRemaining > maxIndex)
+                    {
+                        bytesInThisIteration = maxIndex - currentIndex;
+                        if (bytesInThisIteration == 0)
+                        {
+                            // wraparound; next iteration will get more bytes
+                            currentIndex = 0;
+                            continue;
```

Comment:
1) How can currentIndex being reset to 0 cause invalid data? The entirety of byteBuffer is valid data, right up to the value of maxIndex.  In the scenario you describe, maxIndex is within 1MB of the end of the entirety of byteBuffer.  So everything up to that point, including index 0, must be valid data.  In fact, resetting in that situation is exactly what I want, since the intention is always to loop through the available data in byteBuffer.

2) If currentIndex is 0, and maxIndex is greater than 0 (which it is whenever there is any data to play), then no matter how big bytesRemaining is, we will calculate a positive nonzero value for bytesInThisIteration.  If we do that, we will not take the inner "if", we won't continue, and we won't infinite loop.

I tested this by recording a short (<1MB) sample; it plays back fine, and it fills a 1MB buffer with multiple copies of itself as intended.  So I am not changing the logic here, but I did add some comments.